datanode/entities/position.go

```diff
@@ -111,6 +117,17 @@ func (p *Position) UpdateWithSettleDestressed(e settleDestressed) {
 	p.TxHash = TxHash(e.TxHash())
 }
 
+func (p *Position) UpdateWithSettleMarket(e settleMarket) {
+	markPriceDec := num.DecimalFromUint(e.SettledPrice())
+	openVolumeDec := num.DecimalFromInt64(p.OpenVolume)
+
+	unrealisedPnl := openVolumeDec.Mul(markPriceDec.Sub(p.AverageEntryPrice)).Div(e.PositionFactor())
+	p.RealisedPnl = p.RealisedPnl.Add(unrealisedPnl)
+	p.UnrealisedPnl = decimal.Zero
```

Comment:
Not a fan of this, TBH... Of course we have `type Decimal = decimal.Decimal` in the `num` package, but if we ever were to swap out the underlying type, having the underlying type bleed in to the code base would be an issue

Reply:
that's on the datanode though, it initially started with using the decimal package directly.

Reply:
Thanks, I've updated the other places it was used in the other positions methods too with this commit https://github.com/vegaprotocol/vega/pull/6262/commits/dd3080400fa705880944db06480782ac0a45ee6b. It'll be used a lot in other places in data-node so maybe we should raise a low priority ticket to clean that up at some point.